frontend/src/metabase/embedding/embedding-iframe-sdk-setup/utils/get-common-embed-settings.ts

```diff
@@ -0,0 +1,110 @@
+  SdkIframeQuestionEmbedSettings,
+} from "../types";
+
+const GET_ENABLE_GUEST_EMBED_SETTINGS: (data: {
+  isSimpleEmbedFeatureAvailable: boolean;
+  experience: SdkIframeEmbedSetupExperience;
+}) => SdkIframeEmbedSetupGuestEmbedSettings &
+  Pick<SdkIframeEmbedSetupSettings, "useExistingUserSession"> = ({
+  isSimpleEmbedFeatureAvailable,
+  experience,
+}) => {
+  const isQuestionOrDashboardEmbed =
+    isQuestionOrDashboardExperience(experience);
+
+  return {
+    ...(isQuestionOrDashboardEmbed
+      ? {
+          isGuestEmbed: true,
+          useExistingUserSession: false,
+          ...(isQuestionOrDashboardEmbed && {
+            drills: false,
+            // We force set `downloads` to `true` when the `simple embedding` feature is not enabled (OSS)
+            ...(!isSimpleEmbedFeatureAvailable && {
+              withDownloads: true,
+            }),
+          }),
+        }
+      : {
+          isGuestEmbed: false,
+          useExistingUserSession: true,
+        }),
+    hiddenParameters: [],
+  };
+};
+
+const GET_DISABLE_GUEST_EMBED_SETTINGS: (data: {
+  state:
+    | Pick<
+        SdkIframeEmbedSetupSettings,
+        "isGuestEmbed" | "useExistingUserSession"
+      >
+    | undefined;
+  experience: SdkIframeEmbedSetupExperience;
+}) => SdkIframeEmbedSetupGuestEmbedSettings &
+  Pick<SdkIframeEmbedSetupSettings, "useExistingUserSession"> &
+  Pick<
+    SdkIframeDashboardEmbedSettings | SdkIframeQuestionEmbedSettings,
+    "lockedParameters"
+  > = ({ state, experience }) => {
+  const isQuestionOrDashboardEmbed =
+    isQuestionOrDashboardExperience(experience);
+  const isQuestionEmbed = experience === "chart";
+
+  return {
+    ...(isQuestionOrDashboardEmbed
+      ? {
+          isGuestEmbed: false,
+          useExistingUserSession: state?.useExistingUserSession,
+          drills: true,
+        }
+      : {
+          isGuestEmbed: false,
+          useExistingUserSession: state?.useExistingUserSession,
+        }),
+    ...(isQuestionEmbed && {
+      // Currently, a chart should not have hidden parameters in non-guest embed mode
+      hiddenParameters: [],
+    }),
+  };
+};
+
+export const getCommonEmbedSettings = ({
+  state,
+  experience,
+  isGuestEmbedsEnabled,
+}: {
+  state:
+    | Pick<
+        SdkIframeEmbedSetupSettings,
+        "isGuestEmbed" | "useExistingUserSession"
+      >
+    | undefined;
+  experience: SdkIframeEmbedSetupExperience;
+  isGuestEmbedsEnabled: boolean;
+}) => {
+  const isSimpleEmbedFeatureAvailable =
+    PLUGIN_EMBEDDING_IFRAME_SDK_SETUP.isFeatureEnabled();
+
+  if (isSimpleEmbedFeatureAvailable) {
+    return isGuestEmbedsEnabled && state?.isGuestEmbed
+      ? GET_ENABLE_GUEST_EMBED_SETTINGS({
+          experience,
+          isSimpleEmbedFeatureAvailable,
+        })
+      : GET_DISABLE_GUEST_EMBED_SETTINGS({ state, experience });
+  } else {
+    return GET_ENABLE_GUEST_EMBED_SETTINGS({
+      experience,
+      isSimpleEmbedFeatureAvailable,
+    });
+  }
```

Comment:
nit: simplified conditionals

```suggestion
  const shouldDisable =
    isSimpleEmbedFeatureAvailable &&
    !(isGuestEmbedsEnabled && state?.isGuestEmbed);
  
  return shouldDisable
    ? GET_DISABLE_GUEST_EMBED_SETTINGS({ state, experience })
    : GET_ENABLE_GUEST_EMBED_SETTINGS({ experience, isSimpleEmbedFeatureAvailable });
```